lib/spack/docs/packaging_guide.rst

```diff
@@ -2343,21 +2348,23 @@ requires Python 2, you can similarly leave out the lower bound:
 Notice that we didn't use ``@:3``. Version ranges are *inclusive*, so
 ``@:3`` means "up to and including any 3.x version".
 
-What if a package can only be built with Python 2.7? You might be
-inclined to use:
+If we require Python 3 exactly, we can simply write:
 
 .. code-block:: python
 
-   depends_on('python@2.7')
+   depends_on('python@3')
 
-However, this would be wrong. Spack assumes that all version constraints
-are exact, so it would try to install Python not at ``2.7.18``, but
-exactly at ``2.7``, which is a non-existent version. The correct way to
-specify this would be:
+since we don't care about the latter digits. You can think of this as a
+short-hand for ``@3:3``. Among equivalent specs, usually the most succinct is
+preferred.
 
-.. code-block:: python
+Note that if a version part is alphabetic, it is considered to precede
+numeric versions, and it is compared to other alphabetic sequences using
+dictionary order, unless it is a special name, so:
```

Comment:
We should document which "special names" are accepted. In `lib/spack/spack/version.py`, can you convert `#` to `#:` before `infinity_version`? Then you can link to it here:
```suggestion
dictionary order, unless it is a :py:data:`special name <spack.version.infinity_versions>`, so:
```

Reply:
What is # and #: ?

Reply:
Comment and docstring comment.